### ISSUE TITLE
There should be Best Practices page for Windows containers

### ISSUE THREAD

**Issue by olljanat** [MicrosoftDocs/Virtualization-Documentation#818]
**Description**
As Windows containers/Docker/Kubernetes introduces new features all the time it is very hard to keep track that which features/configurations are really tested to be production ready.

That why I think that we should introduce new Best Practices page which list configurations we know working for sure.

Here is first draft based on my experience:

| Topic | Best Practice | Source/Reason |
| ---------- | ----------- | ----------- |
| OS version | Use latest Windows Server from semi-annual channel | https://goo.gl/yL6iaB |
| Isolation mode | Use process isolation mode | Hyper-V isolation mode adds lot of overhead and contains lot of known issues (e.g. moby/moby#37283 ) |
| Swarm mode | Use parameter *--endpoint-mode dnsrr* for all the services on Swarm (including services provided by Linux nodes) | https://goo.gl/3JK9do / https://goo.gl/rst6Pf |
| Kubernetes | Do not use on production | Experimental feature |
| Build server | Use Core version of Windows Server | https://goo.gl/uDPAqw |
| Swarm manager | Use Linux server on swarm manager role | https://goo.gl/ieiK9V |

**Comment by IvanJosipovic** [comment]
Agreed, its very hard to understand what is supported and what is not.
For example, lots of docs talk about Server 2016, and don't specify Server Core 1709/1803. The support for those two version was just added in Docker EE 18.03.1-ee-1

Another good resource:
https://success.docker.com/article/compatibility-matrix

**Comment by scmikes** [comment]
Thanks for the update.  I also passed your link to our Microsoft Contact.

**Comment by scooley** [comment]
@cwilhit @taylorb-microsoft - does that chart look right to you?  If so I'll formally doc it.

**Comment by olljanat** [comment]
@cwilhit notice that original post already couple of months old. Now when Windows Server 2019 is out LTSC probably is option again. What comes to stability. From containers point of view SAC versions have been lot more stable than Win Srv 2016 which is very buggy on that area.

Dev workstations is only place where I can see that Hyper-V isolation mode can be useful but IMO it is very interesting option that it is now (finally) possible to use process isolation mode on Windows 10 too (look https://github.com/MicrosoftDocs/Virtualization-Documentation/issues/898) . I would not use it even on dev servers and definitely not in production. It is much better idea to setup build process to create Docker images for all needed OS versions. Example of that you can find from https://github.com/olljanat/multi-win-version-aspnet


Anyway, IMO it is most important to have clear documentation that which configurations are tested to be **production** ready and that one should be of course to be updated time to time.

**Comment by olljanat** [comment]
It would be also nice to have documented that if [using host mode in production](https://blogs.technet.microsoft.com/virtualization/2017/09/26/dockers-ingress-routing-mesh-available-with-windows-server-version-1709/) is still recommendation with Windows Server 2019 and latest Docker version?

**Comment by robinharwood** [comment]
We hugely appreciate your contribution, feedback, and support to our technical documentation.

We’ve phased out GitHub Issues on this repository as the feedback mechanism for content and have replaced it with a new feedback system. For more information see, [Content User Feedback](https://aka.ms/ContentUserFeedback). As part of this rollout, we are migrating your issue where our team will use your comments to improve the documentation experience. Unfortunately, this does mean you won't be able to access this issue after the migration is complete.

All issues will be migrated by Thursday 29th August 2024.

Once again, thank you for your valuable feedback, we hope you’ll continue to contribute using our new feedback system.
